addons/website_sale_stock/static/src/js/website_sale_stock.js

```diff
@@ -52,7 +51,10 @@ $('.oe_website_sale').each(function() {
             if (_.isEmpty(_.difference(variant_ids[k][1], values))) {
                 var info = variant_ids[k][4];
                 if(_.contains(['always', 'threshold'], info['inventory_availability'])) {
-                    info['virtual_available'] -= parseInt(info['cart_qty']);
+                    if (info['cart_reduced'] != true){
```

Comment:
@JKE-be 
Now it is a hidden issue as it will not show negative qty again...what I try to do here is to only once reduce the cart_qty as there is no XHR to the backend which will set back the qty to the virtual available in the database...so clicking or adding and removing qty will always reduce the virtual available qty by the cart qty which is wrong...

Reply:
I was decided to try to merge it today, but I don't understand the purpose of this code, and even with a debugger, it seems just useless :/

Can you explain it in more detail and maybe a step to reproduce it.

Thnks

Reply:
Hmm, I think the difference is that I have a JSON object in the ```data-attribute_value_ids```and it is actively referencing, so in my case it saves the values dynamically and reuse it on each triggered event and is reducing the virtual_available amount from the result of the last event every time I increase or reduce the the qty by the whole cart qty which I do circumvent by adding this flag. This is why you can't reproduce it...you have to do it every time again b/c your values are static. Both are working correct with this and the other "fix"

So, it is a consequence of the possibility to have a JSON object in the ```data-attribute_value_ids``` instead of the need to parse it in case it is a string or in case the added dict is not JSON encoded with json.dumps in the backend.

Hmm, IMHO it does not change the behavior of the current code but prevents a misbehavior if the value is a bit different (better IMHO). As said, it is impossible to pass a boolean to the frontend without the check ```_.isString(variant_ids)```, so this does not work without the other commit.

I am not an JS expert, but it feels right what I did and how it does behave now compared to what is currently implemented. I would very much appreciate if you could make it more flexible with this and the other additions.

<details>

<summary>Backend code to understand what is available in the frontend later</summary>

```python

    def get_attribute_value_ids(self, product):
        res = super(WebsiteSaleStockInherit, self).get_attribute_value_ids(product)
        variant_ids = [r[0] for r in res]
        for record, variant in izip(res, request.env['product.product'].sudo().browse(variant_ids)):
            # this sample dict is normally retrieved by another method to add extra information
            extended_info = {
                'some_value': 5,
                'another_value': True,
            }
            record[4].update(extended_info)

        return json.dumps(res)
``` 
</details>

Reply:
@JKE-be 
May I ask you for feedback to my last explanation...I use it in production now for quite a while without side-effects, so it is at least not harmful IMHO.